### PR TITLE
[CALCITE-2939] NPE when executeBatch array type.

### PR DESCRIPTION
**Pull request by bakea**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/avatica/remote/TypedValue.java [hunk 1]
```diff
@@ -359,6 +359,9 @@ private static Object serialToJdbc(ColumnMetaData.Rep type, ColumnMetaData.Rep c
           copy.add(serialToJdbc(componentRep, null, o, calendar));
         }
       }
+      if (componentRep == null && list.size() > 0) {
+        componentRep = ((TypedValue) list.get(0)).type;
+      }
```
Comment (danny0405) [inline core/src/main/java/org/apache/calcite/avatica/remote/TypedValue.java]: The document already say that the componentType in TypedValue is a non-null value [1], the only way is that when the nested parse in [2], which is a null we passed in initiatively.  For null i think it is reasonable cause we do not know what the value component type yet. The code you patched is reasonable if we finally got the list element type.

[1] https://github.com/apache/calcite-avatica/blob/63bab44362c808a8f4981e5d5f5af7d942982f3b/core/src/main/java/org/apache/calcite/avatica/remote/TypedValue.java#L165
[2] https://github.com/apache/calcite-avatica/blob/63bab44362c808a8f4981e5d5f5af7d942982f3b/core/src/main/java/org/apache/calcite/avatica/remote/TypedValue.java#L359

File: core/src/main/java/org/apache/calcite/avatica/remote/TypedValue.java [hunk 2]
```diff
@@ -359,6 +359,9 @@ private static Object serialToJdbc(ColumnMetaData.Rep type, ColumnMetaData.Rep c
           copy.add(serialToJdbc(componentRep, null, o, calendar));
         }
       }
+      if (componentRep == null && list.size() > 0) {
+        componentRep = ((TypedValue) list.get(0)).type;
```
Comment (danny0405) [inline core/src/main/java/org/apache/calcite/avatica/remote/TypedValue.java]: If we still can not got the type, we should throw an exception, can you please add a check here ?

Comment (bakea) [reply core/src/main/java/org/apache/calcite/avatica/remote/TypedValue.java]: I think so.

**Comment by F21** [comment]
@danny0405 Can you look at the updated changes?

**Pull request merged**